index.js

```diff
@@ -72,16 +74,13 @@ function parse(str, options) {
       continue
     }
 
-    var key = str.slice(index, eqIdx).trim()
+    var key = str.slice(index, eqIdx++).trim()
 
     // only assign once
     if (undefined === obj[key]) {
-      var val = str.slice(eqIdx + 1, endIdx).trim()
-
-      // quoted values
-      if (val.charCodeAt(0) === 0x22) {
-        val = val.slice(1, -1)
-      }
+      var val = (str.charCodeAt(eqIdx) === 0x22)
+        ? str.slice(eqIdx + 1, endIdx - 1).trim()
```

Comment:
I like this change for performance, but I think this changes the behavior. Before it trimmed whitespace around potential quotes, but now it'd trim whitespace within quotes.